src/components/Identity/index.js

```diff
@@ -10,30 +10,67 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
-import { defer } from "../../utils";
+import { defer, assign, crc32 } from "../../utils";
 import processIdSyncs from "./processIdSyncs";
+import createEvent from "../DataCollector/createEvent";
+import { validateCustomerIds, normalizeCustomerIds } from "./util";
 
-const ECID_NAMESPACE = "ECID";
+import { COOKIE_NAMES } from "./constants";
+
+const { CUSTOMER_ID_HASH, EXPERIENCE_CLOUD_ID } = COOKIE_NAMES;
 
 const addIdsContext = (payload, ecid) => {
   // TODO: Add customer ids.
-  payload.addIdentity(ECID_NAMESPACE, {
+  payload.addIdentity(EXPERIENCE_CLOUD_ID, {
     id: ecid
   });
 };
 
 const createIdentity = ({ config, logger, cookie }) => {
   // We avoid reading the ECID from the cookie right away, because we
   // need to wait for the user to opt in first.
-  const getEcid = () => cookie.get(ECID_NAMESPACE);
+  const getEcid = () => cookie.get(EXPERIENCE_CLOUD_ID);
   let optIn;
   let responseRequested = false;
   let deferredForEcid;
+  let network;
+  let lifecycle;
+  const customerIds = {};
+
+  const makeServerCall = payload => {
+    return lifecycle
+      .onBeforeDataCollection(payload)
+      .then(() => network.sendRequest(payload, false));
```

Comment:
@Aaronius ,
This line should be 
```
return Promise.resolve()
    .then(() => {
      return lifecycle.onBeforeDataCollection(payload);
    })
    .then(() => {
      return network.sendRequest(payload, false);
    });
```

Reply:
Why?

Reply:
Without this wrapping the the command is not waiting for first interact call to get an ECID

Reply:
It sounds like we're basically getting lucky with using `Promise.resolve` here then and not being intentional in our strategy of sequencing. We need to think this through in more depth. I'll try to dedicate some time to figuring out how to properly solve the problem.

Reply:
@jileeshs I think your problem might be this line: https://github.com/adobe/alloy/pull/129/files#diff-c626ca482d730eb64b741a9d718607adR102 and how it's using the `ecid` variable that was initialized `before` the request for the ECID occurred, and therefore it is undefined. Compare this with the `master` branch, which retrieves the `ecid` again instead.

If this doesn't fix the issue, let me know and I might need you to show me the problem.

Reply:
That didn't work.